### ISSUE TITLE
[BERT] Runtime pre-training data sample generation

### ISSUE THREAD

**Issue by eric-haibin-lin** [dmlc/gluon-nlp#675]
Currently for BERT pre-training, the samples (with masks) have to be generated ahead of time, and usually take gigabytes of disk space. Need to investigate ways to do this on the fly in the training loop without impacting the training throughput.
